### PR TITLE
✨ [#2192] Display newsletter form errors to user

### PR DESCRIPTION
**Pull request by stevenbal**
task: https://taiga.maykinmedia.nl/project/open-inwoner/task/2192

~~Based on #1082, so that should be merged first~~

### CODE REVIEW

File: src/open_inwoner/accounts/views/profile.py [hunk 1]
```diff
@@ -330,6 +330,10 @@ def get_form_kwargs(self) -> dict[str, Any]:
     def form_valid(self, form):
         form.save(self.request)
 
+        # Display errors raised by Laposta API
+        if form.errors:
+            return self.form_invalid(form)
```
Comment (pi-sigma) [inline src/open_inwoner/accounts/views/profile.py:335]: Should be logged before returning?

File: src/open_inwoner/laposta/client.py [hunk 2]
```diff
@@ -33,39 +33,39 @@ def get_lists(self) -> list[LapostaList]:
...
-            return None
+        response = self.delete(f"member/{member_id}", params={"list_id": list_id})
+
+        if response.status_code == 400:
+            data = response.json()
+            error = data.get("error", {})
+            # Handle scenario where a subscription does not exists in the API,
+            # but it does exist locally
+            if error.get("code") == 203 and error.get("parameter") == "member_id":
+                return None
```
Comment (pi-sigma) [inline src/open_inwoner/laposta/client.py:66]: In `create_subscription` there is a logging call in the same spot, here it's missing

File: src/open_inwoner/laposta/forms.py [hunk 3]
```diff
@@ -38,26 +45,45 @@ def __init__(self, user=None, *args, **kwargs):
...
+        to_create = []
+        for list_id in newsletters:
+            if list_id in existing_subscriptions:
+                continue
+
+            try:
+                member = client.create_subscription(list_id, user_data)
+            except (RequestException, ClientError):
+                logger.exception(
+                    "Something went wrong while trying to create subscription"
```
Comment (pi-sigma) [inline src/open_inwoner/laposta/forms.py:75]: Perhaps catch the errros as `exc` or something and add then add `exc_info=exc` to the call

File: src/open_inwoner/laposta/forms.py [hunk 4]
```diff
@@ -67,17 +93,33 @@ def save(self, request, *args, **kwargs):
...
+            user=request.user, list_id__in=unsubscribe_from_ids
+        )
+
+        to_delete_ids = []
+        for subscription in unsubscribe_from:
+            try:
+                client.remove_subscription(subscription.list_id, subscription.member_id)
+            except (RequestException, ClientError):
+                logger.exception(
+                    "Something went wrong while trying to create subscription"
```
Comment (pi-sigma) [inline src/open_inwoner/laposta/forms.py:110]: See previous

**Review by pi-sigma — CHANGES_REQUESTED** [review]

File: src/open_inwoner/laposta/forms.py (hunk 3 above)
Comment (stevenbal) [reply src/open_inwoner/laposta/forms.py:75]: `logging.exception` adds the traceback by default: https://docs.python.org/3/library/logging.html#logging.Logger.exception

File: src/open_inwoner/accounts/views/profile.py (hunk 1 above)
Comment (stevenbal) [reply src/open_inwoner/accounts/views/profile.py:338]: Added a `log_user_action`

**Review by pi-sigma — APPROVED** [review]

File: src/open_inwoner/laposta/forms.py (hunk 4 above)
Comment (Bartvaderkin) [inline src/open_inwoner/laposta/forms.py:110]: This log line mentions create (instead of remove/unsubscribe)

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]
So close: found one log string.

**Review by Bartvaderkin — APPROVED** [review]

**Pull request merged**